functions.py

```diff
@@ -183,19 +183,26 @@ def get_jenkins_jobs(server, job_search_fields):
 	for i in range(fields_length):
 		fields[i] = fields[i].strip(' ')
 
-	# fetch all jobs from server
-	all_jobs = server.get_jobs()
-
-	# parse out all jobs that do not contain any search field and/or are not OSP10, OSP13, OSP15 or OSP16 jobs
+	# check for fields that contain valid regex
 	relevant_jobs = []
 	supported_versions = ['10', '13', '15', '16', '16.1']
-	for job in all_jobs:
-		job_name = job['name']
-		if any(supported_version in job_name for supported_version in supported_versions):
-			for field in fields:
-				if field in job_name:
+	for field in fields:
+		try:
+
+			# does the search field contain a valid regex or search field?
+			re.compile(field)
+
+			# fetch all jobs from server that match the given regex or search
+			all_jobs = server.get_job_info_regex(field)
```

Comment:
@agarcia64 doesn't this give any errors from Jenkins API?

Reply:
If the regex is invalid (as in it is syntactically incorrect) then it will throw the exception error and move onto the next search field, if it exists.

Reply:
So why can't we catch that exception?

Reply:
I'm not sure I know how to catch the exception. re.compile just compiles the given string into a regex object (that we don't save anywhere). I use that function because it'll will throw an error if given invalid regex.

Reply:
I tested it and it gives the same error as re.error would give, ill suggest the code, pls test it

Error compiling regex: missing ), unterminated subpattern at position 5
No jobs found with given search field. Exiting...

Reply:
Ok, sounds good!